test/strategies/Strategy.test.ts

```diff
@@ -158,10 +161,14 @@ export function testStrategy(
     }
 
     after(async function () {
+      // Reset network.
       await network.provider.request({
         method: "hardhat_reset",
         params: [],
       })
+
+      // Reset configs.
+      execSync("git checkout -- ./configs && git clean -fd ./configs")
```

Comment:
interesting, just from curiousity: what went wrong without this line?

Reply:
Without this, any deployment and upgrade of previous test affects on the following tests. Let's say we test 'Cash - Deploy' and then 'Cash - Upgrade After Deploy'. Then, test deployment of 'Cash - Deploy' lingers so that we end up running 'Cash - Upgrade After Deploy' on it. It ends up with an 'there is no contract' error because we start each test with newly forked mainnet.

Reply:
The major cons of this, assuming that all config files are finished ones, is if you're working on new config, then you have to make sure that you committed it before you run test :( 

This design choice is to enable to run one deployment script to deploy as many as investable you want, even portfolios that include any investable that is deployed beforehand in the same script.

Reply:
so what if we set:

useDeployedImplementation = false

in

async function deployProxy(
  Contract: ethers.ContractFactory,
  args: unknown[] = [],
  opts?: {
    initializer?: string | false,
    unsafeAllow?: ValidationError[],
    constructorArgs?: unknown[],
    timeout?: number,
    pollingInterval?: number,
    useDeployedImplementation?: boolean,
    kind?: 'uups' | 'transparent',
  },
): Promise<ethers.Contract>

Should that bypass the the deployment plugin looking into the config?

Reply:
Why and how does it relevant to this issue?